edgedbpkg/edgedb_ls/__init__.py

```diff
@@ -0,0 +1,343 @@
+            elif self.version.is_prerelease():
+                pre = self.version.pre
+                assert pre is not None
+                pre_phase = packages.semver_pre_tag(self.version)
+                return f"{self.version.major}-{pre_phase}{pre.number}"
+            else:
+                return f"{self.version.major}"
+        else:
+            if self.version.text.find(".dev") == -1:
+                return self.base_slot
+            else:
+                return f"{self.base_slot}-dev"
+
+    def version_includes_revision(self) -> bool:
+        return True
+
+    def sh_get_build_wheel_env(
+        self, build: targets.Build, *, site_packages_var: str
+    ) -> dict[str, str]:
+        env = dict(
+            super().sh_get_build_wheel_env(
+                build, site_packages_var=site_packages_var
+            )
+        )
+        if build.target.is_portable():
+            runstate = ""
+        else:
+            runstate = str(build.get_install_path("runstate") / "edgedb")
+
+        env["EDGEDB_BUILD_PACKAGE"] = "1"
+        env["EDGEDB_BUILD_RUNSTATEDIR"] = runstate
+        return env
+
+    def get_build_script(self, build: targets.Build) -> str:
+        common_script = super().get_build_script(build)
+
+        if build.channel == "stable" and not self.version.is_stable():
+            raise AssertionError(
+                f"cannot build non-stable edgedb-ls=={self.version} "
+                f"for the stable channel"
+            )
+
+        return common_script
+
+    def get_build_install_script(self, build: targets.Build) -> str:
+        script = super().get_build_install_script(build)
+        dest = build.get_install_dir(self, relative_to="pkgbuild")
+
+        bindir = build.get_install_path("bin").relative_to("/")
+
+        ep_helper_pkg = build.get_package("pyentrypoint")
+        ep_helper = (
+            build.get_temp_dir(ep_helper_pkg, relative_to="pkgbuild")
+            / "bin"
+            / "pyentrypoint"
+        )
+
+        script += textwrap.dedent(
+            f"""\
+            for p in "{dest}/{bindir}"/*; do
+                if [ -f "$p" ]; then
+                    mv "$p" "${{p}}.py"
+                    chmod -x "${{p}}.py"
+                    sed -i -e "/#!/d" "${{p}}.py"
+                    cp "{ep_helper}" "$p"
+                fi
+            done
+            """
+        )
+
+        return script
+
+    def get_private_libraries(self, build: targets.Build) -> list[str]:
+        return ["libcrypto.*", "libssl.*"]
+
+    def get_exposed_commands(self, build: targets.Build) -> list[pathlib.Path]:
+        bindir = build.get_install_path("bin")
+
+        return [
+            bindir / "edgedb-ls",
+        ]
+
+    def get_conflict_packages(
+        self,
+        build: targets.Build,
+        root_version: str,
+    ) -> list[str]:
+        return ["edgedb-common"]
+
+
+class Cryptography(packages.PythonPackage):
```

Comment:
No need to copy these, import from the main edgedb-server build package instead.